docs/opensearch_querqy_design.md

```diff
@@ -0,0 +1,273 @@
+# OpenSearch Querqy Plugin Design Document
+
```

Comment:
I think we should keep the Design Document as a developer documentation for those who want to understand what's going on under the hood apart from the user documentation and keep the latter at https://docs.querqy.org/querqy/index.html. I would not describe the details of the Common Rules rewriter in the design document (I've written two versions of this documentation and I can tell it's a lot of work ;-) ). The Querqy documentation is maintained here https://github.com/querqy/querqy-docs and can be changed via PRs.

Reply:
This makes sense, I'd remove the usage part from the doc. Will rename this to developer documentation and just keep plugin specific stuff here.